### ISSUE TITLE
Test cases execute more slowly as number of tests in spec increases

### ISSUE THREAD

**Issue by dsodtang** [cypress-io/cypress#18456]
### Current behavior

The amount of time a single test case takes to execute increases with the number of test cases that exist within a spec. This is a problem for a few specs that I maintain due to them having a large number of data-driven test cases.

The increase in run time seems to be worse in tests that have larger numbers of cy.(whatever) calls. All the numbers I've listed below are from the test code included in this report.

Single test run-time w 100 tests containing one cy.wait - between 9ms and 45ms - overall run time of ~2s
Single test run-time w 500 tests containing one cy.wait - between 9ms and 152ms - overall run time of ~27s
Single test run time w 100 tests containing five cy.wait - between 23ms and 72ms - overall run time of ~4s
Single test run time w 500 tests containing five cy.wait - between 23ms and 165ms - overall run time of ~42s
Single test run time w 100 tests containing 25 cy.wait - between 141ms and 186ms - overall run time of ~15s
Single test run time w 500 tests containing 25 cy.wait - between 209ms and 259ms - overall run time of ~1m47s

### Desired behavior

Have consistent execution time for an individual test case no matter how many other test cases are to be run within the same spec.

### Test code to reproduce

This is what I used to reproduce the issue. Increasing NumberOfTests increases the average amount of time each individual test takes

```
describe('this causes problems', () => {
    const NumberOfTests = 100;
    for(let x = 0; x < NumberOfTests ; x++) {
        it('test ' + x, () => {
            cy.wait(1);
            cy.wait(1);
            cy.wait(1);
            cy.wait(1);
            cy.wait(1);
            cy.wrap(true).should('be.true');
        });
    }
});
```

### Cypress Version

8.5.0

### Other

My run time numbers were derived while using the mochawesome reporter. Running without the reporter decreased per-test overhead slightly but did not affect the observed behavior of run time increasing with the number of tests.

**Comment by dsodtang** [comment]
Will retest to keep the bot happy. As far as I know there's been no change here.

**Comment by dsodtang** [comment]
Retest done. Per-test run times appear to be consistent now no matter how many run when using the sample code in the description.

Looks like it has been fixed somewhere between report time and 10.1.0.
Thanks.
